request_reader.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"encoding/json"
+	"io"
+
+	gyaml "github.com/ghodss/yaml"
+	"gopkg.in/yaml.v2"
+)
+
+type requestReader interface {
+	next() ([]byte, error)
+}
+
+// jsonRequestReader parses multiple JSON objects from given reader
+// JSON objects can be delimited by space or newline
+type jsonRequestReader struct{ *json.Decoder }
```

Comment:
I would avoid embedding this here. The methods of `json.Decoder` should not become methods of the jsonRequestReader:

```suggestion
type jsonRequestReader struct{ dec *json.Decoder }
```